test/runtests.jl

```diff
@@ -1,48 +1,37 @@
-using MINLPTests
+using MINLPTests, JuMP, Test
 
-using Test
-using JuMP
+using Ipopt, Juniper
 
-using Ipopt
-using Juniper
-
-ipopt = IpoptSolver(print_level=0)
-juniper = JuniperSolver(IpoptSolver(print_level=0), log_levels=[])
-
-nlp_solvers = [ipopt]
-minlp_solvers = [juniper]
-poly_solvers = [ipopt]
-mipoly_solvers = [juniper]
+const IPOPT = IpoptSolver(print_level=0)
+const JUNIPER = JuniperSolver(IpoptSolver(print_level=0), log_levels=[])
 
+const NLP_SOLVERS = [IPOPT]
+const MINLP_SOLVERS = [JUNIPER]
+const POLY_SOLVERS = [IPOPT]
+const MIPOLY_SOLVERS = [JUNIPER]
 
 @testset "JuMP Model Tests" begin
-
-for solver in nlp_solvers
-    @testset "$(string(typeof(solver))) NLP Tests" begin
-        MINLPTests.test_nlp(solver)
+    @testset "$(string(typeof(solver))): nlp" for solver in NLP_SOLVERS
+        MINLPTests.test_nlp(solver, exclude = [
+            "005_011",  # Uses the function `\`
+            "008_011"   # Requires quadratic duals
+        ])
         MINLPTests.test_nlp_cvx(solver)
     end
-end
-
-for solver in minlp_solvers
-    @testset "$(string(typeof(solver))) MINLP Tests" begin
-        MINLPTests.test_nlp_mi(solver)
+    @testset "$(string(typeof(solver))): nlp_mi" for solver in MINLP_SOLVERS
+        MINLPTests.test_nlp_mi(solver, exclude = [
+            "005_011",  # Uses the function `\`
+            "003_013",  # Bug in Juniper - handling of expression graph?
+            "006_010"   # Bug in Juniper - handling of user-defined functions.
```

Comment:
@ccoffrin are these expected failures for Juniper?

Reply:
Yes, they were failing when I built the dataset, https://github.com/lanl-ansi/MINLPTests.jl/blob/master/src/nlp-mi/tests.jl